packages/widget/src/pages/SettingsPage/ResetSettingsButton.tsx

```diff
@@ -31,29 +50,62 @@ export const ResetSettingsButton: React.FC = () => {
         tools.bridges.map((tool) => tool.key),
         tools.exchanges.map((tool) => tool.key),
       );
+      setDefaultSettings(config);
     }
     toggleDialog();
   };
 
+  const isSlippageChanged = config.slippage
+    ? Number(slippage) !== config.slippage * 100
+    : slippage !== defaultConfigurableSettings.slippage;
+
+  const isRoutePriorityChanged = config.routePriority
+    ? routePriority !== config.routePriority
+    : routePriority !== defaultConfigurableSettings.routePriority;
+
+  const isGasPriceChanged = gasPrice !== defaultConfigurableSettings.gasPrice;
+
+  const isCustomRouteSettings =
+    tools?.bridges?.length !== enabledBridges?.length ||
+    tools?.exchanges?.length !== enabledExchanges?.length ||
+    isSlippageChanged ||
+    isRoutePriorityChanged ||
+    isGasPriceChanged;
+
   return (
-    <Box px={3} mt={1.5} mb={1}>
-      <Button onClick={toggleDialog} fullWidth>
-        {t('button.resetSettings')}
-      </Button>
-      <Dialog open={open} onClose={toggleDialog}>
-        <DialogTitle>{t('warning.title.resetSettings')}</DialogTitle>
-        <DialogContent>
-          <DialogContentText>
-            {t('warning.message.resetSettings')}
-          </DialogContentText>
-        </DialogContent>
-        <DialogActions>
-          <Button onClick={toggleDialog}>{t('button.cancel')}</Button>
-          <Button variant="contained" onClick={handleReset} autoFocus>
-            {t('button.reset')}
+    isCustomRouteSettings && (
+      <Box px={3} mt={1.5} mb={1}>
+        <ResetButtonContainer>
+          <Box display={'flex'} marginBottom={'12px'}>
+            <InfoRounded
+              sx={{
+                marginRight: '8px',
```

Comment:
Just a matter of preference and everywhere in the codebase we use `?:` for rendering or in this case it would be nice to do

```ts
if (!isCustomRouteSettings) {
 return null
}

return (<>...</>)
```
You can also search for [Stop using &&](https://kentcdodds.com/blog/use-ternaries-rather-than-and-and-in-jsx) 😄

Reply:
hahaha, yeah I read this, the one time I use`&&` 😂.
Thanks for pointing this out though !!

fixing this